row.py

```diff
@@ -558,3 +558,83 @@ def summarize_run(folder, run_name):
             "unrecognized characters": len(frame.loc[(frame["Empty Text"] > 0), :].index),
         },
     )
+
+
+def build_mosaic_image(images, object_name):
+    """build a mosaic image from a list of cv2 images
+
+    Args:
+        images (list): list of cv2 images to mosaic together
+        object_name (str): the name of the image object (original filename)
+
+    Returns:
+        mosaic_image (np.ndarray): composite mosaic of smaller images
+    """
+    max_width = 0
+    buffer = 5
+
+    #: Loop through all images to get sizes, save largest
+    for img in images:
+        if img.shape[1] > max_width:
+            max_width = img.shape[1]
+
+    #: Set up parameters for mosaic, calculate number of cols/rows
+    number_images = len(images)
+    number_columns = math.floor(math.sqrt(number_images))
+    number_rows = math.ceil(number_images / number_columns)
+
+    logging.info("mosaicking %i images into %i column by %i row grid", number_images, number_columns, number_rows)
+
+    #: Build mosaic image with white background
+    tile_width = max_width + 2 * buffer
+    total_height = tile_width * number_rows
+    total_width = tile_width * number_columns
+    mosaic_image = np.zeros((total_height, total_width, 3), dtype=np.uint8)
+    mosaic_image[:, :] = (255, 255, 255)
+
+    if total_height * total_width > 40_000_000:
+        logging.info('mosaic image size is too large: "%s"', object_name)
```

Comment:
**issue**: this should be an error so we find it in the logs.

We should split the image into multiple docs but we should see if this condition is reached before spending any time on it.

Reply:
duplicate